### DISCUSSION TITLE
Add a Reset Leave Condition.

### DISCUSSION THREAD

**Discussion by JVKeller** [horilla-opensource/horilla#359] [Q&A]
Hello. Just reviewing this for the first time and it looks very nice and well polished...

I noticed for the leave conditions, you can only set it to a date... How hard would it be to add the employee/users start date (anniversary date) as the renewal date?

Sorry, Typo in the title...  "Reset Leave Conditions"

I'm terrible at coding but I'm playing around with it a bit. If I can get this feature to work, I can use this for sure!

**Comment by JVKeller** [comment]
So, I'm playing around with this...

After I make any changes to the code and push them to the branch I made, I can never seem to get the updates to show.

I even deleted the static files and regenerated.

What commands do I need to run to get the updates showing? 

IT seems I have to delete the entire stack and re-deploy it....

**Reply by horilla-opensource** [reply]
Hi @JVKeller 
If only view changes are made, it'll automatically be reflected on service restart.

If there are changes in the model, you'll have to run the migrations commands.

**Comment by JVKeller** [comment]
Just notating a typo I saw...

horilla/locale/en/LC_MESSAGES/django.po
line 23029 'available'

**Reply by horilla-opensource** [reply]
Thanks for pointing it out.

We'll check and update the po files.

Thank you looking into this areas. We greatly appreciate your efforts.

With Regards,
**Team Horilla**

**Reply by JVKeller** [reply]
Ok. I Think I have it working but won't know until a day goes by I guess.

I do see one error that needs to be addressed but it only shows up when you create the leave type, not when you set it to the anniversary setting. I think because it's still looking for an update interval which goes away after it's saved.


```
ValueError
invalid literal for int() with base 10: ''
/app/leave/models.py, line 423, in set_reset_date
horilla.decorators.wrapped_view
/usr/local/bin/python3.10
3.10.15
['/app',  '/usr/local/bin',  '/usr/local/lib/python310.zip',  '/usr/local/lib/python3.10',  '/usr/local/lib/python3.10/lib-dynload',  '/usr/local/lib/python3.10/site-packages']
```

**Reply by JVKeller** [reply]
@horilla-opensource  I must have missed someting. I set the leave to reset on todays date and was hoping that when I checked it today, it would have reset to the 5 days I have it at, but nothing changed.

**Comment by ghost** [comment]
Any chance on getting this added to the pending release? I'm not having good luck with it. I deleted what I tried and I'll give it one more try.

**Comment by JVKeller** [comment]
Had to work on some other projects, but not I'm back to try and figure this out.

Digging through the code more, I've seen some better opportunities to maybe get this working by using the contract start date, rather than the employee start date. This will also add more flexibility too.

**Comment by JVKeller** [comment]
Alright, I have this about 90% ready!
@horilla-opensource 

Is there an easy way I can test the functions rather than waiting for the next day to see if the leave type was renewed?

**Reply by horilla-opensource** [reply]
Hi @JVKeller ,

You'll have to alter the time of your system to have it checked. We'll try to bring up a call to action for the scheduler where it'll be possible to call a scheduler action for testing.

With Regards,
**Team Horilla**

**Reply by JVKeller** [reply]
@horilla-opensource I had good hopes it would renew, but it didn't seem to work right. Can you point out which file I should be adding the logic for the renewal to?

**Reply by horilla-opensource** [reply]
> @horilla-opensource I had good hopes it would renew, but it didn't seem to work right. Can you point out which file I should be adding the logic for the renewal to?

Hi @JVKeller ,

This is the function for reset the leave type: https://github.com/horilla-opensource/horilla/blob/d464c9e90ba237523b75070b03ff5a987b893e0c/leave/scheduler.py#L11C1-L12C1
